cpp/include/nvtext/tokenize.hpp

```diff
@@ -175,5 +175,42 @@ std::unique_ptr<cudf::column> character_tokenize(
   cudf::strings_column_view const& strings,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Creates a strings column from a strings column of tokens and an
+ * associated column of row ids.
+ *
+ * Multiple tokens from the input column may be combined into a single row (string)
+ * in the output column. The tokens are concatenated along with the `separator` string
+ * in the order in which they appear in the `row_indices` column.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ["hello", "world", "one", "two", "three"]
+ * r = [0, 0, 1, 1, 1]
+ * s1 = detokenize(s,r)
+ * s1 is now ["hello world", "one two three"]
+ * r = [0, 2, 1, 1, 0]
+ * s2 = detokenize(s,r)
+ * s2 is now ["hello three", "one two", "world"]
+ * @endcode
+ *
+ * All null row entries are ignored and the output contains all valid rows.
+ *
+ * @throw cudf::logic_error is `separator` is invalid
+ * @throw cudf::logic_error if `row_indices.size() != strings.size()`
+ * @throw cudf::logic_error if `row_indices` contains nulls
```

Comment:
What happens if an index is negative?

Reply:
Negative index would be invalid and would likely crash.

Reply:
We can easily/efficiently detect this in `token_row_offsets_fn`. I would prefer to have both the comment and the check.

Reply:
I don't see how to check the data without launching a kernel.
I'd rather libcudf not check for bad input data per discussion at #5505